src/libstore/ssh.cc

```diff
@@ -109,6 +125,8 @@ Path SSHMaster::startMaster()
     ProcessOptions options;
     options.dieWithParent = false;
 
+    logger->pause();
```

Comment:
Here the logger won't be resumed if there is an exception. So it's better to have `pause()` return a RAII helper object whose destructor calls `resume()`. Or use `Finally cleanup = [&]() { logger->resume(); }`.